sklearn/metrics/tests/test_classification.py

```diff
@@ -127,6 +128,30 @@ def test_multilabel_accuracy_score_subset_accuracy():
         assert_equal(accuracy_score(y2, [(), ()], normalize=False), 0)
 
 
+def test_balanced_accuracy_score():
+    """Test balanced accuracy score for binary classification task"""
+
+    # test on imbalanced data set
+    y_true = np.array([0, 1, 1, 1, 1, 1, 1, 1, 1, 1])
+    y_pred = np.array([1, 1, 1, 1, 1, 1, 1, 1, 1, 1])
+
+    assert_equal(balanced_accuracy_score(y_true, y_pred), 0.5)
+
+    # test the function with the equation defined as
+    # 0.5*true positives/(true positives + false negatives) +
+    # 0.5*true negatives/(true negatives + false positives)
+    y_true, y_pred, _ = make_prediction(binary=True)
+    tn, fp, fn, tp = np.bincount(y_true * 2 + y_pred, minlength=4)
+    bas = 0.5 * tp / (tp + fn) + 0.5 * tn / (tn + fp)
+    assert_equal(balanced_accuracy_score(y_true, y_pred), bas)
+
+    # test using non-binary
```

Comment:
This is still a binary classification problem, but with string class labels instead of integers class labels.